src/libraries/Microsoft.Extensions.DependencyModel/src/Microsoft.Extensions.DependencyModel.ref.cs

```diff
@@ -51,8 +49,8 @@ public partial class CompilationOptions
         private readonly object _dummy;
         private readonly int _dummyPrimitive;
         public Dependency(string name, string version) { throw null; }
-        public readonly string Name { get { throw null; } }
-        public readonly string Version { get { throw null; } }
+        public string Name { get { throw null; } }
```

Comment:
Revert

Reply:
Is this actually a Roslyn reference assembly bug? ILSpy shows me the following in the reference assembly:

```
[System.Runtime.CompilerServices.NullableContext(1)]
[System.Runtime.CompilerServices.Nullable(0)]
public readonly struct Dependency : IEquatable<Dependency>
{
	public string Name
	{
		[CompilerGenerated]
		get
		{
			throw null;
		}
	}
```

Reply:
@eerhardt actually what we had in the reference source isn't valid C#:

![image](https://user-images.githubusercontent.com/7412651/155409778-e5467e56-2a9a-448d-8c3e-b3a0e344cd6b.png)

Reply:
> what we had in the reference source isn't valid C#:

I find that hard to believe, considering that we are compiling the reference source.

You are using a `class` in that snippet, this is on a `struct`.

Reply:
> ILSpy shows me the following in the reference assembly:

Notice the `readonly` in the signature: `public readonly struct Dependency`.

https://github.com/dotnet/runtime/blob/385fc8832b1d4160abe13645054b65458b6a12f8/src/libraries/Microsoft.Extensions.DependencyModel/src/Dependency.cs#L10

Reply:
Got it. Another GenAPI bug that needs to be investigated.